### PR TITLE
[MAINTENANCE] Confirm E2E workflow for Cloud-enabled `DataContext` integration with `DataContextVariables`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Minor changes to logic to ensure proper `GET` and `PUT` requests are made.
- E2E tests to ensure coverage over changes.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: azure-pipelines-dependency-graph-testing.yml [hunk 1]
```diff
@@ -48,6 +48,10 @@ resources:
 
 variables:
   GE_USAGE_STATISTICS_URL: "https://qa.stats.greatexpectations.io/great_expectations/v1/usage_statistics"
+  GE_CLOUD_BASE_URL: $(GE_CLOUD_BASE_URL)
+  GE_CLOUD_ACCESS_TOKEN: $(GE_CLOUD_ACCESS_TOKEN)
+  GE_CLOUD_ORGANIZATION_ID: $(GE_CLOUD_ORGANIZATION_ID)
```
Comment (cdkini) [inline azure-pipelines-dependency-graph-testing.yml]: To be set using the Azure UI

File: great_expectations/data_context/data_context/base_data_context.py [hunk 2]
```diff
@@ -302,7 +302,7 @@ def __init__(
         # necessary properties / overrides
         self._synchronize_self_with_underlying_data_context()
 
-        self._variables = self._init_variables()
+        self._variables = self._data_context.variables
```
Comment (cdkini) [inline great_expectations/data_context/data_context/base_data_context.py:305]: Bugfix to ensure we're instantiating `CloudDataContextVariables`

File: great_expectations/data_context/data_context_variables.py [hunk 3]
```diff
@@ -106,12 +106,12 @@ def _get(self, attr: DataContextVariableSchema) -> Any:
...
         return substituted_val
 
-    def save_config(self) -> None:
+    def save_config(self) -> Any:
         """
         Persist any changes made to variables utilizing the configured Store.
         """
         key: ConfigurationIdentifier = self.get_key()
-        self.store.set(key=key, value=self.config)
+        return self.store.set(key=key, value=self.config)
```
Comment (cdkini) [inline great_expectations/data_context/data_context_variables.py:114]: Small change so we can assert that `self.store.set` returns `True` in a cloud environment - useful to check that the endpoint is functional.

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 4]
```diff
@@ -169,16 +169,17 @@ def _update(self, ge_cloud_id: str, value: Any, **kwargs: dict) -> bool:
...
             f"organizations/"
             f"{organization_id}/"
-            f"{hyphen(self.ge_cloud_resource_name)}/"
-            f"{ge_cloud_id}",
+            f"{hyphen(self.ge_cloud_resource_name)}",
         )
+
+        if ge_cloud_id:
+            data["data"]["id"] = ge_cloud_id
+            url = urljoin(f"{url}/", ge_cloud_id)
```
Comment (cdkini) [inline great_expectations/data_context/store/ge_cloud_store_backend.py:181]: `/endpoint` vs `/endpoint/` makes a difference!

File: tests/conftest.py [hunk 5]
```diff
@@ -2333,6 +2333,22 @@ def empty_ge_cloud_data_context_config(
...
+    """
+    base_url = os.environ["GE_CLOUD_BASE_URL"]
+    organization_id = os.environ["GE_CLOUD_ORGANIZATION_ID"]
+    access_token = os.environ["GE_CLOUD_ACCESS_TOKEN"]
+    ge_cloud_config = GeCloudConfig(
+        base_url=base_url,
+        organization_id=organization_id,
+        access_token=access_token,
+    )
+    return ge_cloud_config
```
Comment (cdkini) [inline tests/conftest.py:2349]: Should we x-fail if we don't have creds? Not 100% how to treat this.

File: tests/data_context/test_data_context_variables.py [hunk 6]
```diff
@@ -505,3 +531,69 @@ def test_file_data_context_variables_e2e(
         file_data_context.variables.plugins_directory == value_associated_with_env_var
     )
     assert config_saved_to_disk.plugins_directory == f"${env_var_name}"
+
+
+@pytest.mark.integration
+@pytest.mark.cloud
```
Comment (cdkini) [inline tests/data_context/test_data_context_variables.py:535]: Do we care about this mark? Currently used in the `dgtest-cloud-overrides` section of the `dependency_graph` pipeline to ensure we always test our Cloud integrations - if we want to keep this, let's be consistent and ensure we mark other tests the same way.

File: script [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+export GE_CLOUD_BASE_URL="https://api.dev.greatexpectations.io"
+export GE_CLOUD_ORGANIZATION_ID="0ccac18e-7631-4bdd-8a42-3c35cce574c6"
+export GE_CLOUD_ACCESS_TOKEN="6284c351faa74ee6bcfcc6cfede25aac"
```
Comment (Kilo59) [inline script:5]: Should this be committed?

Comment (cdkini) [inline script]: ughhhhh no 😭

**Review by anthonyburdi — APPROVED** [review]
LGTM!

File: tests/conftest.py (hunk 5 above)
Comment (anthonyburdi) [reply tests/conftest.py:2349]: No, I think we fail if this test does not run well for any reason incl unreachable cloud. Although we may consider moving these e2e tests to a separate integration test pipeline that doesn't run on each PR.

File: tests/data_context/test_data_context_variables.py [hunk 8]
```diff
@@ -1,12 +1,16 @@
 import copy
-import os
```
Comment (Kilo59) [inline tests/data_context/test_data_context_variables.py:2]: 🎊

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**